packages/grafana-ui/src/components/Combobox/Combobox.tsx

```diff
@@ -69,17 +78,29 @@ function itemFilter<T extends string | number>(inputValue: string) {
 export const Combobox = <T extends string | number>({
   options,
   onChange,
-  value,
+  value: valueProp,
   isClearable = false,
   createCustomValue = false,
   id,
   width,
   'aria-labelledby': ariaLabelledBy,
   ...restProps
 }: ComboboxProps<T>) => {
-  const [items, setItems] = useState(options);
+  // Value can be an actual scalar Value (string or number), or an Option (value + label), so
+  // get a consistent Value from it
+  const value = typeof valueProp === 'object' ? valueProp?.value : valueProp;
+
+  const isAsync = typeof options === 'function';
+  const loadOptions = useLatestAsyncCall(isAsync ? options : asyncNoop); // loadOptions isn't called at all if not async
```

Comment:
So I kinda thought we could actually make `loadOptions` be always called. Like instead of having async/sync mode to always be async, just with static options it would me much quicker by using `Promise.resolve(options)`.

Looking at the code I am not 100% this would be more readable or easy to do. For example the `selectedItem` value would have to be redone to handle async. Loading may flicker so may need a bit of a debunce or something.

The benefits though I think would be that there is a single mode of operation. So for example right now from the code is not clear to me how `value` prop will work in case it is async? I sort of assume it gets maybe handled as a custom value from `onInputValueChange`? 

So just a suggestion at this point but I feel making this always async could deal with some possible discrepancies between async/sync modes.

Reply:
Yeah, I think that could be an approach to fix some of the rough edges. I think we can explorer that seperately (along with all the other minor bugs with this 😓)

Reply:
@tskarhed basically you would do something like this:
```
useLatestAsyncCall(typeof options === 'function' ? options : () => Promise.resolve(options))
```

At that point you don't have to think about whether the combobox is used async or not, it will always be async. You don't care if the async call actually takes time or it's resolved at the next exec loop tick, you now handle both cases the same way. So the basic stuff is by default harder (because you cannot just access options as a static array) but then you have to solve every feature just for the async case (which you should anyway) so overall it should be less work and everything should work the same whether you supply a static array of options or a async function.